packages/nakama-js/socket.ts

```diff
@@ -353,18 +355,19 @@ export interface SocketError {
 
 /** A socket connection to Nakama server implemented with the DOM's WebSocket API. */
 export class DefaultSocket implements Socket {
-  private socket?: WebSocket;
   private readonly cIds: { [key: string]: PromiseExecutor };
   private nextCid: number;
 
   constructor(
       readonly host: string,
       readonly port: string,
       readonly useSSL: boolean = false,
-      public verbose: boolean = false) {
+      public verbose: boolean = false,
+      readonly adapter : WebSocketAdapter = new WebSocketAdapterText()
```

Comment:
Do we want the adapter to be accessible directly from `DefaultSocket` type?

Reply:
@novabyte do you mean having the user pass an enum or something here instead of the adapter? And then not exporting the adapter directly.